plugins/microchip/switchtec-nvme.c

```diff
@@ -261,37 +287,141 @@ static int pax_get_nvme_pf_functions(struct pax_nvme_device *pax,
 	for (i = 0; i < pax_all.ep_port_all.ep_port_count; i++) {
 		ep_port = &pax_all.ep_port_all.ep_ports[i];
 		if (ep_port->port_hdr.type == SWITCHTEC_GFMS_DB_TYPE_EP) {
-			n = ep_port->ep_ep.ep_hdr.function_number;
+			ep = &ep_port->ep_ep;
+			ret = pax_enum_ep(ep, functions + index,
+					  max_functions - index);
+			index += ret;
+		} else if (ep_port->port_hdr.type ==
+			   SWITCHTEC_GFMS_DB_TYPE_SWITCH) {
+			n = ep_port->port_hdr.ep_count;
+
 			for (j = 0; j < n; j++) {
-				function = &ep_port->ep_ep.functions[j];
-				if (function->sriov_cap_pf == CAP_PF
-				    && function->device_class == NVME_CLASS)
-					memcpy(&functions[index++],
-						function,
-						sizeof(*function));
+				ep = ep_port->ep_switch.switch_eps + j;
+				ret = pax_enum_ep(ep, functions + index,
+						  max_functions - index);
+
+				index += ret;
 			}
 		}
 	}
 
 	return index;
 }
 
+static int pax_build_nvme_pf_list(struct pax_nvme_device *pax,
+				  struct switchtec_gfms_db_ep_port_attached_device_function *functions,
```

Comment:
Sorry for the very long struct name, we can indent less (just one tab) in this case, like below, to make it less than 80-char a line.
```
static int pax_get_nvme_pf_functions(struct pax_nvme_device *pax,
        struct switchtec_gfms_db_ep_port_attached_device_function *functions,
        ...
```